src/main/java/uk/gov/hmcts/darts/audio/service/impl/MediaRequestServiceImpl.java

```diff
@@ -500,4 +513,59 @@ public List<AdminMediaSearchResponseItem> adminMediaSearch(Integer transformedMe
             && mediaEntity.getEnd().isAfter(mediaRequest.getStartTime())).toList();
         return AdminMediaSearchResponseMapper.createResponseItemList(filteredMediaList, hearing);
     }
+
+    @Override
+    public MediaHideResponse adminHideOrShowMediaById(Integer mediaId, MediaHideRequest mediaHideRequest) {
+        MediaHideResponse response;
+
+        IdRequest<MediaHideRequest> request = new IdRequest<>(mediaHideRequest, mediaId);
+        mediaHideOrShowValidator.validate(request);
+
+        Optional<MediaEntity> mediaEntityOptional
+            = mediaRepository.findById(mediaId);
+        if (mediaEntityOptional.isPresent()) {
+            MediaEntity mediaEntity = mediaEntityOptional.get();
+
+            mediaEntity.setHidden(mediaHideRequest.getIsHidden());
+            mediaEntity = mediaRepository.saveAndFlush(mediaEntity);
```

Comment:
If this succeeds, but the later operations (such as creating/deleting the Reason) fail, will this be rolled back, or would we be left in an inconsistent state? Wondering if we need to wrap `adminHideOrShowMediaById()` in `@Transactional`.